src/autocomplete.js

```diff
@@ -145,32 +148,44 @@ class Autocomplete {
         this.stickySelectionTimer.cancel();
         this.stickySelection = false;
     }
-
+    seen(completion) {
+        if (!this.hasSeen.has(completion) && completion && completion.completer && completion.completer.onSeen && typeof completion.completer.onSeen === "function") {
```

Comment:
Could you prefix the method name with $ to indicate that it is a private method, and move `has` function call to the end  condition chain. 

Also do we not allow `?.` syntax?

Reply:
We don't use `?.` anywhere in the code base.